scripts/metrics-migration/migration.py

```diff
@@ -0,0 +1,640 @@
+#!/usr/bin/env python3
+"""
+VictoriaMetrics migration script.
+
+This script exports metrics from VictoriaMetrics, allows transformation,
+and imports them back to VictoriaMetrics with streaming support.
+"""
+
+import argparse
+import gzip
+import io
+import json
+import os
+import sys
+import threading
+from pathlib import Path
+from typing import Callable, Optional
+
+import requests
+from tqdm import tqdm
+
+# Check environment variable to skip SSL certificate verification
+SKIP_SSL_VERIFY = os.getenv("SKIP_SSL_VERIFY", "").lower() in ("1", "true", "yes")
+
+
+def identity_transform(line: dict) -> dict:
+    """Default transformation function that returns the line unchanged."""
+    return line
+
+
```

Comment:
Let's add this function to DRY many others soon:
```suggestion

def auto_open(file_name: str, write: bool = False):
    open_fn = gzip.open if file_name.lower().endswith(".gz") else open
    return open_fn(file_name, "wt" if write else "rt", encoding="utf-8")


```

Reply:
This is less readable than the current implementation.